core/src/main/scala/chisel3/StrongEnum.scala

```diff
@@ -275,13 +279,13 @@ abstract class EnumFactory {
   protected def do_Value(name: String): Type = {
     val result = new Type
 
-    // We have to use UnknownWidth here, because we don't actually know what the final width will be
-    result.bindToLiteral(id, UnknownWidth())
-
-    enumRecords.append(EnumRecord(result, name))
+    val resId = if (!oneHot) id else BigInt(1 << id.toInt)
```

Comment:
```suggestion
    val resId = if (!oneHot) id else BigInt(1) << id.toInt
```
You need to make sure to left shift the `BigInt`, otherwise you're left shifting the `Int` which can overflow.

Reply:
Thanks Jack!